### PR TITLE
Misc Commands

### LINKED ISSUE #3: Miscellaneous Commands
## Hard coded
- [x] help
- [x] reminder
- [x] code block
- [x] ping
- [x] dice
- [x] Setup command

## Guild
- [x] info
  - [x] ombuds
  - [x] stuvo
  - [x] pal
- [x] notas
- [x] rooster

## Canvas
- [x] get courses/module/items (lesmateriaal)
  - [x] User friendly

### PR DESCRIPTION
**Pull request by xunafay**
pull request for #3

### CODE REVIEW

File: cfg/config.json [hunk 1]
```diff
@@ -1,7 +1,44 @@
 {
     "discord": 
         {
-            "prefix": "!"
+            "prefix": "!",
+
+            "richpresence":
+            {
+                "statusType": "2",
+                "interval": "60000",
```
Comment (xunafay) [inline cfg/config.json]: can be of data type int

File: src/discord.ts [hunk 2]
```diff
@@ -7,6 +7,26 @@ export async function buildClient(): Promise<Client> {
   const client = new Client();
   client.on('ready', () => {
     console.log(`Logged in as ${client.user?.tag}`);
+
+    /*
+      Rich presence updating.
+      Value may not be below 15000 (rate-limit discord api).
+    */
+    const interval = (parseInt(data.discord.richpresence.interval) < 15000 ? 15000 : parseInt(data.discord.richpresence.interval));
```
Comment (xunafay) [inline src/discord.ts]: use Math.max

File: src/checkreminders.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+
+//import { promises } from "fs";
+
+import { Client, TextChannel } from 'discord.js';
+
+//const getguild = (guildID: string, client: Client) => Promise.resolve(client.guilds?.fetch(guildID));
+
+
+export function checkreminderinit(client: Client): void {
```
Comment (xunafay) [inline src/checkreminders.ts]: use camel case when naming functions

File: src/checkreminders.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+
+//import { promises } from "fs";
```
Comment (xunafay) [inline src/checkreminders.ts]: remove

File: src/checkreminders.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+import { Client, TextChannel } from 'discord.js';
+
+//const getguild = (guildID: string, client: Client) => Promise.resolve(client.guilds?.fetch(guildID));
```
Comment (xunafay) [inline src/checkreminders.ts]: remove

File: src/checkreminders.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+
+export function checkreminderinit(client: Client): void {
+  setInterval(async function () {
+
+    //currently reffers to the TM testing ground + #test (channel)
+    const guildID = '780572565240414208'; //TODO get guild id from db to post in correct server
+    const channelID = '782283497582624778'; //TODO get channel of the reminder from the database
+
+    const guild = await Promise.resolve(client.guilds?.fetch(guildID));
```
Comment (xunafay) [inline src/checkreminders.ts]: `.fetch()` already returns a promise no need to wrap it in another promise

File: src/checkreminders.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    const channel: TextChannel = await Promise.resolve(guild.client.channels.fetch(channelID)) as TextChannel;
```
Comment (xunafay) [inline src/checkreminders.ts]: `.fetch()` already returns a promise no need to wrap it in another promise
[validate channel is a text channel](https://discord.js.org/#/docs/main/stable/class/Channel?scrollTo=type) before casting

File: src/commands.ts [hunk 8]
```diff
@@ -65,5 +67,46 @@ export const commands: Command[] = [
...
+        return getNotes(message.channel.id.toString());
+      }
+      //When incorrectly used (includes !notes help)
+      else {
+        return new Formatter()
+          .bold('Help for '+command(guildConfig.prefix+'notes'),true)
+          .command(guildConfig.prefix+'notes').text(': get notes from your current channel', true)
+          .command(guildConfig.prefix+'notes #channel').text(': get notes from your favourite channel', true)
+          .command(guildConfig.prefix+'notes #channel an amazing note').text(': Enter a note in a channel', true)
+          .build();
```
Comment (DrunkenToast) [inline src/commands.ts]: I don't like the else if chaining.

File: src/commands.ts [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+  {
+    name: 'help',
+    description: 'that\'s this command.',
+    aliases: [],
+    response(message: Message, guildConfig: any): string | MessageEmbedOptions {
+      const help: MessageEmbedOptions = {'title': 'Help is on the way!',
+        'description': commands.concat(guildConfig.commands).map( c => '`' + guildConfig.prefix + c.name + '`: ' + c.description).join('\n'),
+        'color': '43B581',
+        'url': 'http://mydoc.epic',
+        'footer': {text: 'Some commands support putting \'help\' behind it.'}};
```
Comment (xunafay) [inline src/commands.ts]: do not quote keys when unnecessary
format properly

File: src/commands.ts [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+      function getNotes(channelID: string) {
+        //TODO send something else when channel doesnt have notes (rn simply undefined)
+        let i = 0;
+        const embed: MessageEmbedOptions = {'title': 'Notes',
+          'description': 'Notes for channel <#'+channelID+'>:\n'
+          +guildConfig.notes[channelID]?.map((note: string) => ++i + ' • ' + note).join('\n'),
+          'footer': {text: 'For help: '+guildConfig.prefix+'notes help'}
+        };
+        return embed;
+      }
```
Comment (xunafay) [inline src/commands.ts]: do not define functions within functions
format properly

**Review by xunafay — COMMENTED** [review]
use string interpolation to improve readability

**Comment by PandaGamersHDNL** [comment]
+ added some comments to tokenizer

**Review by PandaGamersHDNL — APPROVED** [review]

**Review by DrunkenToast — APPROVED** [review]

**Pull request merged**